tests/js/server/aql/aql-optimizer-rule-remove-filter-covered-by-index.js

```diff
@@ -287,14 +287,14 @@ function optimizerRuleTestSuite() {
         var result;
         result = AQL_EXPLAIN(query, { }, paramIndexRangeFilter);
         assertEqual([ IndexesRule, FilterRemoveRule ], 
-          removeAlwaysOnClusterRules(result.plan.rules), query);
+          removeAlwaysOnClusterRules(result.plan.rules).filter((r) => r !== "splice-subqueries"), query);
```

Comment:
Would it make sense to add something like `removeAlwaysOnRules`, which we could use everywhere instead of the `.filter((r) => r !== "splice-subqueries")`? It's fine with me to leave it as it is though, we can always add it if we get another "always on" rule that affectes a lot of tests.